lib/matplotlib/type1font.py

```diff
@@ -35,9 +34,234 @@
+    def value(self):
+        return self.raw == 'true'
+
+
+class _KeywordToken(_Token):
+    def is_keyword(self, *names):
+        return self.raw in names
+
+    def value(self):
+        return self.raw
+
+
+class _DelimiterToken(_Token):
+    def is_delim(self):
+        return True
+
+    def opposite(self):
+        return {'[': ']', ']': '[',
+                '{': '}', '}': '{',
+                '<<': '>>', '>>': '<<'
+                }[self.raw]
+
+
+class _WhitespaceToken(_Token):
+    pass
+
+
+class _StringToken(_Token):
+    def value(self):
+        return self.raw[1:-1]
+
+
+class _BinaryToken(_Token):
+    def value(self):
+        return self.raw[1:]
+
+
+class _NumberToken(_Token):
+    def is_number(self):
+        return True
+
+    def value(self):
+        if '.' not in self.raw:
+            return int(self.raw)
+        else:
+            return float(self.raw)
+
+
+def _tokenize(data: bytes, skip_ws: bool):
+    """
+    A generator that produces _Token instances from Type-1 font code.
+
+    The consumer of the generator may send an integer to the tokenizer
+    to indicate that the next token should be _BinaryToken of the given
+    length.
+
+    Parameters
+    ----------
+    data : bytes
+        The data of the font to tokenize.
+
+    skip_ws : bool
+        If true, the generator will drop any _WhitespaceTokens from the output.
+    """
+
+    text = data.decode('ascii', 'replace')
+    whitespace_or_comment_re = re.compile(r'[\0\t\r\014\n ]+|%[^\r\n\v]*')
+    token_re = re.compile(r'/{0,2}[^]\0\t\r\v\n ()<>{}/%[]+')
+    instring_re = re.compile(r'[()\\]')
+    pos = 0
+    next_binary = None
+
+    while pos < len(text):
+        if next_binary is not None:
+            n = next_binary
+            next_binary = (yield _BinaryToken(pos, data[pos:pos+n]))
+            pos += n
+            continue
+        match = whitespace_or_comment_re.match(text, pos)
+        if match:
+            if not skip_ws:
+                next_binary = (yield _WhitespaceToken(pos, match.group()))
+            pos = match.end()
+        elif text[pos] == '(':
+            start = pos
+            pos += 1
+            depth = 1
+            while depth:
+                match = instring_re.search(text, pos)
+                if match is None:
+                    return
+                pos = match.end()
+                if match.group() == '(':
+                    depth += 1
+                elif match.group() == ')':
+                    depth -= 1
+                else:  # a backslash - skip the next character
+                    pos += 1
+            next_binary = (yield _StringToken(start, text[start:pos]))
+        elif text[pos:pos + 2] in ('<<', '>>'):
+            next_binary = (yield _DelimiterToken(pos, text[pos:pos + 2]))
+            pos += 2
+        elif text[pos] == '<':
+            start = pos
+            pos = text.index('>', pos)
+            next_bin = (yield _StringToken(pos, text[start:pos]))
```

Comment:
should be next_binary?  should I be worried that the typo (if it is one) was not caught?

Reply:
You're right, it's a typo, and it wasn't caught because in actual font data binary tokens can only occur after some specific keywords (but the font can define those keywords). I wanted to keep that information out of the tokenizer, but I guess I'll need to add some test cases.